### ISSUE TITLE
aws-sdk-ruby fails on presignedPut for s3 gateway.

### ISSUE THREAD

**Issue by poornas** [minio/mint#149]
Needs investigation. This was run with latest  mint image and minio release build in gateway mode

```
{"name":"aws-sdk-ruby","function":"presignedPut(bucket_name,file_name)","args":{"bucket_name":"384f85c7f53e","file_name":"datafile-1-MB"},"duration":2067.5,"error":"Expected to be created object does NOT exist","status":"FAIL"}

```

**Comment by ebozduman** [comment]
Nitish,
This looks like the same issue you currently worked on, issue#615.
In this case, the server console error message is:
```
cause=Signature does not match 
source=[gateway-handlers.go:295:gatewayAPIHandlers.PutObjectHandler()]
```

**Comment by nitisht** [comment]
@ebozduman the fix for minio-java#615 is now available in minio repo. Can you please verify against `minio/minio:edge` if this is working now?

**Comment by nitisht** [comment]
Verified against `minio/minio:edge` tests passing. So closing this. 

```
{"name":"aws-sdk-ruby","function":"listBuckets()","args":{},"duration":8128.58,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"listObjects(bucket_name)","args":{"bucket_name":"071b46331ef3"},"duration":7746.16,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"makeBucket(bucket_name)","args":{"bucket_name":"071b46331ef3"},"duration":2038.19,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"bucketExists?(bucket_name)","args":{"bucket_name":"non-existing-bucket"},"duration":544.7,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"removeBucket(bucket_name)","args":{"bucket_name":"071b46331ef3"},"duration":1688.43,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"putObject(bucket_name,file)","args":{"bucket_name":"071b46331ef3","file":"/mint/data/datafile-1-MB"},"duration":4386.08,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"removeObject(bucket_name,file)","args":{"bucket_name":"071b46331ef3","file":"/mint/data/datafile-1-MB"},"duration":3645.96,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"getObject(bucket_name,file,destination)","args":{"bucket_name":"071b46331ef3","file":"/mint/data/datafile-1-MB","destination":"/tmp"},"duration":13184.42,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"copyObject(source_bucket_name,target_bucket_name,source_file_name,target_file_name)","args":{"source_bucket_name":"071b46331ef3","target_bucket_name":"0f734d91831d","source_file_name":"datafile-1-MB","target_file_name":""},"duration":10114.48,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"copyObject(source_bucket_name,target_bucket_name,source_file_name,target_file_name)","args":{"source_bucket_name":"071b46331ef3","target_bucket_name":"0f734d91831d","source_file_name":"datafile-1-MB","target_file_name":"datafile-1-MB-copy"},"duration":7549.97,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"presignedGet(bucket_name,file_name)","args":{"bucket_name":"071b46331ef3","file_name":"datafile-1-MB"},"duration":3934.79,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"presignedPut(bucket_name,file_name)","args":{"bucket_name":"071b46331ef3","file_name":"datafile-1-MB"},"duration":6822.51,"status":"PASS"}
{"name":"aws-sdk-ruby","function":"presignedPost(bucket_name,file_name,expires_in_sec,max_byte_size)","args":{"bucket_name":"071b46331ef3","file_name":"datafile-1-MB","expires_in_sec":60,"max_byte_size":3145728},"duration":8907.43,"status":"PASS"}
```
